Doc/library/multiprocessing.rst

```diff
@@ -2332,6 +2332,28 @@ The following example demonstrates the use of a pool::
            result = pool.apply_async(time.sleep, (10,))
            print(result.get(timeout=1))        # raises multiprocessing.TimeoutError
 
+There is also a :class:`ThreadPool` class, which is a subclass of :class:`Pool`
```

Comment:
Two minor suggestions:

* Add a reference to this class in the `multiprocessing.dummy` section.
* Add a reference to the `concurrent.futures` module or thread-specific-APIs as these are more robust APIs in general for what this class offers and unfortunatelly not that many people know about them.

Reply:
> Add a reference to this class in the multiprocessing.dummy section.

Do you think that this should
1. move to the `multiprocessing.dummy` section in its entirety,
2. stay in the "Process Pools" section and be linked to from the `multiprocessing.dummy` section, or
3. go into a new "Thread Pools" section (and if so, where should the new section be placed in the document)?

> Add a reference to the `concurrent.futures` module

Seems reasonable to make a suggestion that guides people towards `concurrent.futures`, but that module currently isn't referenced by the `multiprocessing` docs at all. Are you thinking that should be scoped to just `multiprocessing.Pool.ThreadPool` vs `concurrent.futures.ThreadPoolExecutor`, or should it also mention `multiprocessing.Pool` vs `concurrent.futures.ProcessPoolExecutor`?

Perhaps let's refine the wording here before I update the PR. Are you thinking something like:

> Most users should instead prefer to use a `concurrent.futures.Executor`, which provides a simpler interface to thread pools and process pools than the one provided here. These are usually easier to use correctly, and other modules (like `asyncio`) have builtin support for handling the `concurrent.futures.Future` instances that these pools return for tracking the progress of jobs that are run asynchronously.

Is that about it, or are there more specific points I should make?

> or thread-specific-APIs

I'm not sure what you mean by this bit.

Reply:
> Do you think that this should

I don't mind that much if is on `multiprocessing.dummy` or if `multiprocessing.dummy` refers to this but I would prefer option 1. The reason is that I don't think is a good idea to have a "Thread Pools" section because this is still the multiprocessing module and not only the APIs have been designed without threads initially in mind but also is slightly confusing to users (why is not on "threading"?).

> Are you thinking that should be scoped to just multiprocessing.Pool.ThreadPool vs concurrent.futures.ThreadPoolExecutor

Just scoped for `multiprocessing.Pool.ThreadPool` vs `concurrent.futures.ThreadPoolExecutor`. The reason is that `multiprocessing.Pool.ThreadPool` doesn't have the most friendly API and suffers from some of the design that is process-specific. It also doesn't share some of the functionality (like using `maxtasksperchild`). IIRC, this was partially why this class was never fully documented.

> I'm not sure what you mean by this bit.

Sorry, that was poorly written. I meant to reference https://docs.python.org/3/library/concurrent.futures.html#threadpoolexecutor.